### PR TITLE
LG-3544: Link inline prop

### PR DESCRIPTION
**Pull request by spark33**
## ✍️ Proposed changes

🎟 _Jira ticket:_ [LG-3544](https://jira.mongodb.org/browse/LG-3544)

### For bug fixes, new features & breaking changes

- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] I have run `yarn changeset` and documented my changes

### For new components

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README
- [ ] I have verified the Live Example looks as intended on the design website.

### CODE REVIEW

File: packages/typography/src/Link/Link.styles.ts [hunk 1]
```diff
@@ -94,3 +94,8 @@ export const openInNewTabStyles = css`
   left: -1px;
   height: 12px;
 `;
+
+export const inlineStyles = css`
+  font-size: inherit;
+  line-height: inherit;
```
Comment (bruugey) [inline packages/typography/src/Link/Link.styles.ts:100]: should this have an explicitly display: inline property as well?

Comment (bruugey) [reply packages/typography/src/Link/Link.styles.ts:100]: `inheritSize`?

File: packages/typography/README.md [hunk 2]
```diff
@@ -176,6 +176,7 @@ All props extend the HTMLElementProps of their root tag, however the below compo
 | `hideExternalIcon` | `boolean`                        | Hides the external icon when the current host name is different from the host of the destination URL                                                                                     | `false`                                       |
 | `darkMode`         | `boolean`                        | Determines if the component renders in dark mode                                                                                                                                         | `false`                                       |
 | `as`               | `keyof JSX.IntrinsicElements`    | Element to render the component as. **Note**: This will supersede the behavior of any other props.                                                                                       | `'a'`                                         |
+| `inline`           | `boolean`                        | Determines if the Link should be inherit font size from the parent                                                                                                                       | `false`                                       |
```
Comment (TheSonOfThomp) [inline packages/typography/README.md:179]: This feels like it's going to apply `display: inline`. Prop should be something like `inheritFontSize`

File: packages/typography/src/Link/Link.styles.ts [hunk 3]
```diff
@@ -94,3 +94,8 @@ export const openInNewTabStyles = css`
   left: -1px;
   height: 12px;
 `;
+
+export const inheritSizeStyles = css`
+  font-size: inherit;
+  line-height: inherit;
```
Comment (TheSonOfThomp) [inline packages/typography/src/Link/Link.styles.ts:100]: I still feel like this should be the default behavior. A link should always be the size of whatever it's inside.
If someone wants to use a Link outside of something that has a font-size set, they can always either provide a `baseFontSize` prop, or add custom CSS

**Review by TheSonOfThomp — CHANGES_REQUESTED** [review]
The prop name definitely needs to change.

I also think this should be the default behavior, not toggled with a prop

**Pull request closed without merging**